embedded-tests/src/test/java/org/apache/druid/testing/embedded/k8s/K3sClusterWithOperatorResource.java

```diff
@@ -0,0 +1,313 @@
+
+
+  public K3sClusterWithOperatorResource()
+  {
+    super();
+    manifestFiles.add(Resources.getFileForResource(RBAC_MANIFEST));
+    manifestFiles.add(Resources.getFileForResource(OPERATOR_NAMESPACE_MANIFEST));
+  }
+
+  @Override
+  public K3sClusterWithOperatorResource usingTestImage()
+  {
+    return usingDruidImage(DruidContainerResource.getTestDruidImageName());
+  }
+
+  @Override
+  public K3sClusterWithOperatorResource usingDruidImage(String druidImageName)
+  {
+    this.druidImageName = druidImageName;
+    return this;
+  }
+
+  @Override
+  public K3sClusterWithOperatorResource addService(K3sDruidService service)
+  {
+    services.add(service);
+    return this;
+  }
+
+  @Override
+  public void onStarted(EmbeddedDruidCluster cluster)
+  {
+    client = new KubernetesClientBuilder()
+        .withConfig(Config.fromKubeconfig(getContainer().getKubeConfigYaml()))
+        .build();
+    closer.register(client);
+
+    loadLocalDockerImageIntoContainer(druidImageName, cluster.getTestFolder());
+
+    manifestFiles.forEach(this::applyManifest);
+
+    installHelm(cluster);
+    setupOperatorWithHelm();
+
+    client.pods().inNamespace(OPERATOR_NAMESPACE).resources().forEach(this::waitUntilPodIsReady);
+
+    final Properties commonProperties = new Properties();
+    commonProperties.putAll(cluster.getCommonProperties());
+    commonProperties.remove("druid.extensions.modulesForEmbeddedTests");
+    applyConfigMap(
+        newConfigMap(COMMON_CONFIG_MAP, commonProperties, "common.runtime.properties")
+    );
+
+    initializeDruidTestFolders(cluster.getTestFolder());
+
+    String commonPropertiesString = prepareCommonPropertiesString(commonProperties, cluster.getTestFolder());
+    
+    for (K3sDruidService druidService : services) {
+      applyManifest(druidService.withCommonProperties(commonPropertiesString));
+    }
+
+    client.pods().inNamespace(DRUID_NAMESPACE).resources().forEach(this::waitUntilPodIsReady);
+    services.forEach(this::waitUntilServiceIsHealthy);
+  }
+
+  private void initializeDruidTestFolders(TestFolder testFolder) {
+    testFolder.getOrCreateFolder("druid-storage");
+    testFolder.getOrCreateFolder("druid-storage/segments");
+    testFolder.getOrCreateFolder("druid-storage/segment-cache");
+    testFolder.getOrCreateFolder("druid-storage/metadata");
+    testFolder.getOrCreateFolder("druid-storage/indexing-logs");
+  }
+
+  private String prepareCommonPropertiesString(Properties properties, TestFolder testFolder) {
+    StringBuilder sb = new StringBuilder();
+    sb.append("druid.metadata.storage.type=derby\n");
+    sb.append("    druid.metadata.storage.connector.connectURI=jdbc:derby://localhost:1527/var/druid/metadata.db;create=true\n");
+    sb.append("    druid.metadata.storage.connector.host=localhost\n");
+    sb.append("    druid.metadata.storage.connector.port=1527\n");
+    sb.append("    druid.metadata.storage.connector.createTables=true\n");
```

Comment:
the embedded cluster being used in the new test is already configured to use Psql metadata store. We should remove these configs.

Reply:
I agree, however we need to send some common properties to the yaml to ensure that operator loads some pods. I suggest we let these be. (no, empty properties is not working)